src/main/java/io/fabric8/maven/docker/service/QueryService.java

```diff
@@ -52,7 +53,7 @@ public Container getContainer(String containerId) throws DockerAccessException {
      * @throws DockerAccessException in case of an remote error
      */
     public Container getContainerByName(final String containerName) throws DockerAccessException {
-        for (Container el : docker.listContainers(CONTAINER_LIMIT)) {
+        for (Container el : docker.listContainers(containerLimit)) {
```

Comment:
I changed now `getContainerByName()` and `getContainerForImage()` to not use `listContainers()`. containerByName should work always, getImageForContainer only for API > 1.23 (Docker 1.11). As a fallback for getContainerForImage (which actually is only used by docker:stop and docker:watch afais) the only list() mechanism is used.

The limit was there to prevent unexpected runtime behaviour if you have thousands of containers running. And since the list of containers is sorted according to creation time, I thought that it would be a good idea to impose a limit here (assuming the d-m-p is used for tests and development only with no long running containers started).

But you are right, after the change a limit is even less important, so, if you don't mind I'm going to remove it altogether.